amr-wind/utilities/sampling/Sampling.cpp

```diff
@@ -136,11 +136,11 @@ void Sampling::update_sampling_locations()
 {
     BL_PROFILE("amr-wind::Sampling::update_sampling_locations");
 
-    for (const auto& obj : m_samplers) {
-        obj->update_sampling_locations();
+    if (std::any_of(m_samplers.begin(), m_samplers.end(), [](const auto& obj) {
+            return obj->update_sampling_locations();
+        })) {
+        update_container();
```

Comment:
I could be wrong, but it seems like update_container is overkill for when the positions of the samplers change. Some samplers change the number of points, and update_containers may be required for those, but it seems off that the particle container would need to be set up from scratch every time the positions change. but maybe that distinction could be put in another PR (e.g., flag that differentiates between samplers that are static, samplers that do not change number of points but change positions, and samplers that change both).

Reply:
You are absolutely right. Trying to keep these PRs small. But that is another area of improvement.